numba/parfor.py

```diff
@@ -1009,6 +1022,83 @@ def find_mask_from_size(size_var):
                     for l in body_labels:
                         blocks.pop(l)
 
+    def _replace_loop_access_indices(self, loop_body, index_set, new_index):
+        """
+        Replace array access indices in a loop body with a new index.
+        index_set has all the variables that are equivalent to loop index.
+        """
+        # treat new index like others since replacing it with itself is ok
+        index_set.add(new_index.name)
+
+        # add dummy return to last block for CFG computation
+        # max is last block since we add it manually for prange
+        last_label = max(loop_body.keys())
+        loop_body[last_label].body.append(
+            ir.Return(0, ir.Loc("parfors_dummy", -1)))
+
+        labels = find_topo_order(loop_body)
+
+        # remove dummy return
+        loop_body[last_label].body.pop()
+
+        first_label = labels[0]
+        added_indices = set()
+
+        for l in labels:
```

Comment:
Documentation before this block saying what it is doing.

Reply:
done.